pkgs/applications/misc/dupeguru/default.nix

```diff
@@ -0,0 +1,67 @@
+{stdenv, python3Packages, gettext, qt5, fetchFromGitHub}:
+
+python3Packages.buildPythonApplication rec {
+  pname = "dupeguru";
+  version = "4.0.4";
+
+  src = fetchFromGitHub {
+    owner = "arsenetar";
+    repo = "dupeguru";
+    rev = "${version}";
+    sha256 = "0ma4f1c6vmpz8gi4sdy43x1ik7wh42wayvk1iq520d3i714kfcpy";
+    fetchSubmodules = true;
+  };
+
+  nativeBuildInputs = [ 
+    gettext 
+    python3Packages.setuptools
+    python3Packages.pyqt5
+    qt5.wrapQtAppsHook
+  ];
+
+  pythonPath = with python3Packages; [ 
+    pyqt5
+    send2trash
+    sphinx
+    polib
+    hsaudiotag3k  
+  ];
+
+  format = "other";
+
+  buildFlags = [
+    "i18n"
+    "modules"
+    "qt/dg_rc.py"
+    "pyc"
+  ];
+
+
+  installFlags = [
+    "PREFIX=${placeholder ''out''}"
+    "--assume-old=all"
+    "--assume-old=pyc"
+  ];
+
+  postInstall = ''
+    chmod +x $out/share/dupeguru/run.py
+  '';
+
+  doCheck = false;
+
+  #postFixup = ''
+  #  wrapQtApp $out/bin/dupeguru
+  #'';
+
+  #dontWrapQtApps = true;
+
+  #preFixup = ''
+  #  makeWrapperArgs+=("''${gappsWrapperArgs[@]}")
```

Comment:
```
  dontWrapQtApps = true;

  preFixup = ''
    makeWrapperArgs+=("''${qtWrapperArgs[@]}")
  '';
```

This should be sufficient to prevent double wrapping.

Reply:
Okay, the issue is that the GUI executable is in in `share`, the thing in `bin` is just a symlink. And Python infrastructure only sets up things in `bin`:

https://github.com/NixOS/nixpkgs/blob/8d5e90bdb32c7f8708a5b75eaaca00551e2f1d6b/pkgs/development/interpreters/python/mk-python-derivation.nix#L149
https://github.com/NixOS/nixpkgs/blob/8d5e90bdb32c7f8708a5b75eaaca00551e2f1d6b/pkgs/development/interpreters/python/wrap.sh#L4

Adding

```
postFixup = ''
  wrapPythonProgramsIn "$out/share/dupeguru/run.py" "$out $pythonPath"
'';

should fix that.

Reply:
Hmm, there is a bug in the Python wrapper, this line

https://github.com/NixOS/nixpkgs/blob/8d5e90bdb32c7f8708a5b75eaaca00551e2f1d6b/pkgs/development/interpreters/python/wrap.sh#L84

needs to be 

to work around it, the `preFixup` section should go like

```
    # TODO: A bug in python wrapper
    # see 
    makeWrapperArgs="''${qtWrapperArgs[@]}"
```

Reply:
Sorry, should have been `"$out/share/dupeguru"` not `"$out/share/dupeguru/run.py"`.

Reply:
Fixing this

Reply:
I take that back, testing this change on my laptop is impossible, @worldofpeace tag!